### DISCUSSION TITLE
Response Headers for Expiration, Age

### DISCUSSION THREAD

**Discussion by lpw** [RobinTail/express-zod-api#1586] [Q&A] [answered]
How do developers using express-zod-api handle expiration or age in their APIs?  Some of my endpoints have data that should expire quickly, and some is more static.  I don't see an examples of that in documentation or the usual places.

**Comment by RobinTail** [comment, marked as answer]
I presume you're talking about this header:
https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Age

You can set headers either from a Middleware or from ResultHandler, @lpw 

The handlers of those entities have `response` argument, so you can do `response.set()`:
https://expressjs.com/en/api.html#res.set


```ts
const shortTermFactory = defaultEndpointsFactory
  .addMiddleware(
    createMiddleware({
      input: z.object({}),
      middleware: async ({ response }) => {
        response.set("age", 30);
        return {};
      }
    })
  );
```

The endpoints produced on `shortTermFactory` are going to have the header set.

**Comment by lpw** [comment]
Sorry for the delay in response - this worked great - thanks again!
